tests/functional/test_install.py

```diff
@@ -1026,3 +1026,25 @@ def test_double_install_fail(script, data):
     msg = ("Double requirement given: pip==7.1.2 (already in pip==*, "
            "name='pip')")
     assert msg in result.stderr
+
+
+@pytest.mark.network
+def test_update_to_develop(script, tmpdir):
+    """
+    Test updating from install to develop mode removes original package
+    """
+    path = local_repo(
+        'git+http://github.com/pypa/pip-test-package.git',
+        tmpdir.join("cache"),
+    )
```

Comment:
I'd rather not have a network related test if we can avoid it.
You could use an existing module like https://github.com/pypa/pip/blob/6dc28c1/tests/data/src/singlemodule#L799-L805
Then:
  - install it via its path
  - install it via editable
  - check that pip freeze/list report it as editable
  - uninstall it
  - check that all installation disappeared